lerobot/scripts/push_dataset_to_hub.py

```diff
@@ -209,6 +212,7 @@ def push_dataset_to_hub(
         save_meta_data(info, stats, episode_data_index, meta_data_dir)
 
     if not dry_run:
+        # FIXME: token needs to be a str | None
```

Comment:
cc @Cadene It seems `token` should be a `str | None` according toe the `Dataset.push_to_hub` API. Could this be a problem? How should @WaelKarkoub address this?

Reply:
Nice finding. I will take care of it in another PR.

Reply:
Thanks @Cadene. @WaelKarkoub in that case could you please change to
```suggestion
        # TODO(rcadene): token needs to be a str | None
```